### PR TITLE
LTD-3515: Ignore invalid countersignatures

### PR DESCRIPTION
**Pull request by saruniitr**
### Aim

Countersignatures when they are created initially by countersigning managers are `valid`.
If a countersigning manager rejects countersigning then the Case is sent back to caseworker to edit their recommendation. Depending on the comments they update their recommendation and send it again for countersigning. At this point (after editing) we mark all previous countersignatures as `invalid` because it will start the journey again.
So we need to ignore invalid countersignatures otherwise we end up showing warnings etc even for invalid countersignatures.

 - Refactored a method that generates countersignatures to make it clear on what is happening and easy to parameterize in other tests
 - Add tests that check for presence warning message that the case is sent back when countersignatures are `valid` and the usual warning about case requiring countersignatures when they are `invalid`
 - Use `mock_gov_lu_user` where possible instead of mocking it in every test

[LTD-3515](https://uktrade.atlassian.net/browse/LTD-3515)

### CODE REVIEW

File: caseworker/advice/templates/advice/advice_details.html [hunk 1]
```diff
@@ -89,7 +89,7 @@ <h2 class="govuk-heading-m app-bg-colour--red govuk-!-padding-2">
                 <div class="countersignatures">
                     {% for countersign_advice in case|countersignatures_for_advice:advice  %}
                         {%  if countersign_advice.0.outcome_accepted %}
-                        <div class="govuk-!-padding-4 govuk-!-margin-bottom-2 countersigned-by">
+                        <div class="govuk-!-padding-4 govuk-!-margin-bottom-2 countersigned-by" style="border: 1px solid black;">
```
Comment (saruniitr) [inline caseworker/advice/templates/advice/advice_details.html:92]: noticed during testing that this was missing

File: unit_tests/caseworker/conftest.py [hunk 2]
```diff
@@ -891,55 +891,62 @@ def final_advice(current_user, lu_team):
     }
 
 
-def countersignatures_for_advice(all_advice, accepted=[True]):
-    def first_countersignature(advice):
+def countersignatures_for_advice(all_advice, data):
```
Comment (saruniitr) [inline unit_tests/caseworker/conftest.py:894]: same method as before, just made it more clear on what is happening and easy to parameterize in various tests

File: caseworker/advice/templates/advice/advice_details.html (hunk 1 above)
Comment (antroy-madetech) [reply caseworker/advice/templates/advice/advice_details.html:92]: (prob need to rebuild your assets)

File: caseworker/advice/views.py [hunk 3]
```diff
@@ -121,7 +121,7 @@ def rejected_countersign_advice(self):
         rejection (case will be returned to edit the advice once a rejection has occurred.
         """
         for cs in self.case.get("countersign_advice", []):
-            if not cs["outcome_accepted"]:
+            if cs["valid"] is True and cs["outcome_accepted"] is False:
```
Comment (antroy-madetech) [inline caseworker/advice/views.py:124]: `if cs["valid"] and not cs["outcome_accepted"]:` would be easier to read IMO.

**Review by hnryjmes — APPROVED** [review]
looks good. appreciate some of the cleaning up you have done here as well.

File: unit_tests/caseworker/advice/views/test_countersign_view.py [hunk 4]
```diff
@@ -158,7 +168,11 @@ def test_lu_rejected_senior_countersignature(
...
     data_standard_case["case"]["countersign_advice"] = countersignatures_for_advice(
-        [final_advice], accepted=[True, False]
+        [final_advice],
+        [
+            {"order": FIRST_COUNTERSIGN, "outcome_accepted": True},
+            {"order": SECOND_COUNTERSIGN, "outcome_accepted": False},
+        ],
     )
     requests_mock.get(client._build_absolute_uri(f"/cases/{case_id}"), json=data_standard_case)
     mock_get_gov_user.return_value = (
```
Comment (antroy-madetech) [inline unit_tests/caseworker/advice/views/test_countersign_view.py:178]: We should also test this with some `"valid": False` countersignatures in there as well to prove out the negative logic. Looks like you already catered for this in the rework of the countersignatures_for_advice function.

File: caseworker/advice/templates/advice/advice_details.html (hunk 1 above)
Comment (saruniitr) [reply caseworker/advice/templates/advice/advice_details.html:92]: yep, that was it. Rebuilt assets and it is fine without this change, reverting it.

File: unit_tests/caseworker/advice/views/test_countersign_view.py (hunk 4 above)
Comment (saruniitr) [reply unit_tests/caseworker/advice/views/test_countersign_view.py:178]: In case of `valid=False` we cannot really assert on anything because there won't be any warning.
So it will be just a check to see there is no warning, is that ok?

**Review by antroy-madetech — APPROVED** [review]
LGTM
